In [40]:
#IMPORTATION DES LIBRAIRIES
import os
import pandas as pd

In [41]:
#CHARGEMENT DES DONNEES 
df_ville = pd.read_csv("Cities.csv", sep=",")

#SUPPRESSION DES COLONNES INUTILES
df_ville = df_ville.drop(columns=["insee_code", "city_code", "label", "department_name", "region_name"])
df_ville.head()

zip_code   latitude  longitude department_number      region_geojson_name
0     25650  46.999873   6.498147                25  Bourgogne-Franche-Comté
1     25640  47.361512   6.235167                25  Bourgogne-Franche-Comté
2     25310  47.368384   6.871415                25  Bourgogne-Franche-Comté
3     25240  46.713906   6.265831                25  Bourgogne-Franche-Comté
4     25170  47.228558   5.852187                25  Bourgogne-Franche-Comté

In [42]:
df_caf = pd.read_csv("ndur_s_qf_400_dep.csv", sep=";")
df_caf.head()

Date référence Numéro département     Nom département Numéro région  \
0        2018-07                 53             MAYENNE            52   
1        2018-07                 53             MAYENNE            52   
2        2018-07                 53             MAYENNE            52   
3        2018-07                 54  MEURTHE ET MOSELLE            44   
4        2018-07                 55               MEUSE            44   

         Nom région         Quotient familial  Nombre foyers NDUR  \
0  PAYS DE LA LOIRE  Entre 1200 et 1599 euros                6817   
1  PAYS DE LA LOIRE  Entre 2000 et 3999 euros                1132   
2  PAYS DE LA LOIRE        Entre 4000 ou plus                 133   
3         GRAND EST        Entre 4000 ou plus                 500   
4         GRAND EST        Moins de 400 euros                7166   

   Nombre personnes NDUR  Montant total NDUR  Nombre foyers NDURPAJE  ...  \
0                  26806             2231380                    2565  ...   
1                   4626              154032                     228  ...   
2                    554               11676                      21  ...   
3                   2050               52210                      99  ...   
4                  11093             3511553                     360  ...   

   Nombre foyers NDUREJ  Nombre personnes NDUREJ  Montant total NDUREJ  \
0                  5388                    22677               1061757   
1                  1035                     4340                 96399   
2                   120                      516                  6851   
3                   449                     1896                 22204   
4                  1137                     4053                357994   

   Nombre foyers NDURAL  Nombre personnes NDURAL  Montant total NDURAL  \
0                    53                      117                  6338   
1                     1                        2                   336   
2                     0                        0                     0   
3                     0                        0                     0   
4                  4545                     7190               1145289   

   Nombre foyers NDURINS  Nombre personnes NDURINS  Montant total NDURINS  \
0                    149                       363                  18295   
1                      1                         2                    118   
2                      1                         2                    111   
3                      2                         5                    793   
4                   4433                      7006                1767349   

                 Lieu résidence  
0      PAYS DE LA LOIRE/MAYENNE  
1      PAYS DE LA LOIRE/MAYENNE  
2      PAYS DE LA LOIRE/MAYENNE  
3  GRAND EST/MEURTHE ET MOSELLE  
4               GRAND EST/MEUSE  

[5 rows x 22 columns]

In [43]:
#CHECK FOR MISSING VALUES
print(df_ville.isnull().sum())

zip_code                 0
latitude               211
longitude              211
department_number        0
region_geojson_name      0
dtype: int64


In [44]:
print(df_caf.isnull().sum())

Date référence                  0
Numéro département              0
Nom département              1410
Numéro région                   0
Nom région                   2186
Quotient familial               0
Nombre foyers NDUR              0
Nombre personnes NDUR           0
Montant total NDUR              0
Nombre foyers NDURPAJE          0
Nombre personnes NDURPAJE       0
Montant total NDURPAJE          0
Nombre foyers NDUREJ            0
Nombre personnes NDUREJ         0
Montant total NDUREJ            0
Nombre foyers NDURAL            0
Nombre personnes NDURAL         0
Montant total NDURAL            0
Nombre foyers NDURINS           0
Nombre personnes NDURINS        0
Montant total NDURINS           0
Lieu résidence               2186
dtype: int64


**La base de données *VILLES(df_villes)* contient 211 données de Latitude et Longitude manquantes**
**La base de données *CAF(df_caf)* contient 1410 noms de départements manquants et 2186 nom de région et lieu de résidence manquants**

J'ai supprimé les données manquantes de la base VILLES mais j'ai gardé les données manquantes de la base de données CAF

In [45]:
#SUPPRESSION DES VALEURS MANQUANTES
df_ville = df_ville.dropna()

In [46]:
#Renommer les colonnes
df_caf.rename(columns={"Date référence": "date", 
                                "Numéro département": "num_dep", 
                                "Nom département": "nom_dep", 
                                "Numéro région": "num_region", 
                                "Nom région": "nom_region", 
                                "Quotient familial": "qt_familial", 
                                "Nombre foyers NDUR": "nb_foyers_presta_legale", 
                                "Nombre personnes NDUR": "nb_pers_presta_legale", 
                                "Montant total NDUR": "mt_total_presta_legale",
                                "Nombre foyers NDURPAJE": "nb_foyers_petite_enfance",
                                "Nombre personnes NDURPAJE": "nb_pers_petite_enfance",
                                "Montant total NDURPAJE": "mt_total_petite_enfance",
                                "Nombre foyers NDUREJ": "nb_foyers_enfance_jeunesse",
                                "Nombre personnes NDUREJ": "nb_pers_enfance_jeunesse",
                                "Montant total NDUREJ": "mt_total_enfance_jeunesse",
                                "Nombre foyers NDURAL": "nb_foyers_apl",
                                "Nombre personnes NDURAL": "nb_pers_apl",
                                "Montant total NDURAL": "mt_total_apl",
                                "Nombre foyers NDURINS": "nb_foyers_solidarite_insertion",
                                "Nombre personnes NDURINS": "nb_pers_solidarite_insertion",
                                "Montant total NDURINS": "mt_total_solidarite_insertion",
                                "Lieu résidence": "lieu_residence"}, inplace=True)

In [47]:
df_caf.head()

date num_dep             nom_dep num_region        nom_region  \
0  2018-07      53             MAYENNE         52  PAYS DE LA LOIRE   
1  2018-07      53             MAYENNE         52  PAYS DE LA LOIRE   
2  2018-07      53             MAYENNE         52  PAYS DE LA LOIRE   
3  2018-07      54  MEURTHE ET MOSELLE         44         GRAND EST   
4  2018-07      55               MEUSE         44         GRAND EST   

                qt_familial  nb_foyers_presta_legale  nb_pers_presta_legale  \
0  Entre 1200 et 1599 euros                     6817                  26806   
1  Entre 2000 et 3999 euros                     1132                   4626   
2        Entre 4000 ou plus                      133                    554   
3        Entre 4000 ou plus                      500                   2050   
4        Moins de 400 euros                     7166                  11093   

   mt_total_presta_legale  nb_foyers_petite_enfance  ...  \
0                 2231380                      2565  ...   
1                  154032                       228  ...   
2                   11676                        21  ...   
3                   52210                        99  ...   
4                 3511553                       360  ...   

   nb_foyers_enfance_jeunesse  nb_pers_enfance_jeunesse  \
0                        5388                     22677   
1                        1035                      4340   
2                         120                       516   
3                         449                      1896   
4                        1137                      4053   

   mt_total_enfance_jeunesse  nb_foyers_apl  nb_pers_apl  mt_total_apl  \
0                    1061757             53          117          6338   
1                      96399              1            2           336   
2                       6851              0            0             0   
3                      22204              0            0             0   
4                     357994           4545         7190       1145289   

   nb_foyers_solidarite_insertion  nb_pers_solidarite_insertion  \
0                             149                           363   
1                               1                             2   
2                               1                             2   
3                               2                             5   
4                            4433                          7006   

   mt_total_solidarite_insertion                lieu_residence  
0                          18295      PAYS DE LA LOIRE/MAYENNE  
1                            118      PAYS DE LA LOIRE/MAYENNE  
2                            111      PAYS DE LA LOIRE/MAYENNE  
3                            793  GRAND EST/MEURTHE ET MOSELLE  
4                        1767349               GRAND EST/MEUSE  

[5 rows x 22 columns]

In [48]:
#EXPERTION DES DONNEES AU FORMAT EXCEL POUR VISUALISATION AVEC TABLEAU
df_ville.to_excel("Cities_cleaned.xlsx", index=False)
df_caf.to_excel("CAF_cleaned.xlsx", index=False)

#EXPERTION DES DONNEES AU FORMAT CSV POUR ANALYSE ET REQUETATGE AVEC SPARK SQL
df_caf.to_csv("CAF_cleanedForSQL.csv", index=False)